js/main.js

```diff
@@ -40,5 +47,82 @@ function submitInfo(event) {
   $entryForm.reset();
 }
 
+// function to render entries
+function renderEntry(entry) {
+  // creates elements of DOM tree:
+  const $row = document.createElement('div');
+  $row.className = 'row entry-row';
+  const $column = document.createElement('div');
+  $column.className = 'column-full entry-column';
+  const $entryContainer = document.createElement('div');
+  $entryContainer.className = 'entry-container';
+  const $list = document.createElement('ul');
+  $list.className = 'journal-entry';
+  const $listItem = document.createElement('li');
+  $listItem.className = 'user-entry';
+  const $img = document.createElement('img');
+  $img.setAttribute('src', entry.url);
+  $img.className = 'entry-img';
+  const $textContainer = document.createElement('div');
+  $textContainer.className = 'text-container';
+  const $title = document.createElement('h2');
+  $title.className = 'entry-title';
+  $title.textContent = entry.title;
+  const $notes = document.createElement('p');
+  $notes.className = 'entry-notes';
+  $notes.textContent = entry.notes;
+
+  // append DOM nodes to build tree
+
+  $row.appendChild($column);
+  $column.appendChild($entryContainer);
+  $entryContainer.appendChild($list);
+  $list.appendChild($listItem);
+  $listItem.appendChild($textContainer);
+  $listItem.appendChild($img);
+  $textContainer.appendChild($title);
+  $textContainer.appendChild($notes);
+
+  // prepends DOM tree to list
+  $entriesList.appendChild($row);
+}
+
+// looping function for renderEntry function - iterates nextEntryId
+function arrayLoop(array) {
+  for (let i = 0; i < array.length; i++) {
+    const $entriesArray = array[i];
+    // eslint-disable-next-line no-unused-vars
+    const $renderedEntry = renderEntry($entriesArray);
+  }
+}
+
+// toggles "no entries" text as appropriate:
+function toggleNoEntries(event) {
+  const $entriesArray = data.entries;
+  if ($entriesArray.length > 0) {
+    $noEntryMessage.classList.add('hidden');
+  }
+}
+
 // submit event listener
+$entryForm.addEventListener('submit', toggleNoEntries);
 $entryForm.addEventListener('submit', submitInfo);
+
+// listener for DOMContentLoaded event which calls multiple functions
+document.addEventListener('DOMContentLoaded', function (event) {
+  arrayLoop(data.entries);
+
+  // viewSwap();
+  toggleNoEntries();
+});
+
+// view swapping function
+function viewSwap() {
+  for (const child of $parent.children) {
+    child.classList.toggle('hidden');
+  }
+}
```

Comment:
`viewSwap` should expect a string indiciating which view the user wants to switch to. Depending on that string, the element with the matching `data-view` attribute should either be hidden or shown.

Reply:
Can you give me a hint for this implementation? Struggling with this one.

Reply:
```
function viewSwap(viewName) {
 if (viewName === 'entry-form') {
    // Shows the `div` with `data-view="entry-form"
   // Hides the other `div` with `data-view="entries"`
 } else {
   // does the opposite
}
```